api_client.go

```diff
@@ -246,9 +246,11 @@ func NewAPIClient(cfg *Configuration) (*APIClient, error) {
 	c.cfg = cfg
 	c.Context = GetContext(cfg)
 	c.common.client = c
-	err := GetDefaultHeaders(c)
-	if err != nil {
-		return nil, err
+	if !c.cfg.SkipSessionAuth {
+		err := GetDefaultHeaders(c)
```

Comment:
Why we don't need to GetDefaultHeaders for token based auth?

Reply:
`SkipSessionAuth` is an independent parameter, it is just an option to skip the extra session create roundtrip for envs that do not want it.

Reply:
Thanks, my previous understanding is wrong. So we will step into GetDefaultHeaders for token based auth since the parameter is false by default.